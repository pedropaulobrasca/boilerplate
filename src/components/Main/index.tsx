import * as S from './styles'

const Main = () => {
  return (
    <S.Wrapper>
      <S.Logo
        src={'/img/logo.svg'}
        alt={'Imagem de um atomo e react avancado escrito ao lado.'}
      />
      <S.Title>React Avancado</S.Title>
      <S.Description>
        TypeScript, ReactJS, NextJS, Styled Components
      </S.Description>
      <S.Illustration
        src={'/img/hero-illustration.svg'}
        alt={'Um desenvolvedor de frente pra uma tela com codigos'}
      />
    </S.Wrapper>
  )
}

export default Main
