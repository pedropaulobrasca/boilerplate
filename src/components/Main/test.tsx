import { render, screen } from '@testing-library/react'
import Main from './index'

describe('<Main />', () => {
  it('should render the h1 element', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', {
        name: /react avancado/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
