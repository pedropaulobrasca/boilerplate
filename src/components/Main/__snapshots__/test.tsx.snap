// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the h1 element 1`] = `
<main
  class="styles__Wrapper-sc-dzx1ao-0 jOBvCg"
>
  <img
    alt="Imagem de um atomo e react avancado escrito ao lado."
    class="styles__Logo-sc-dzx1ao-1 epziVg"
    src="/img/logo.svg"
  />
  <h1
    class="styles__Title-sc-dzx1ao-2 cIwDMv"
  >
    React Avancado
  </h1>
  <h2
    class="styles__Description-sc-dzx1ao-3 emthCT"
  >
    TypeScript, ReactJS, NextJS, Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente pra uma tela com codigos"
    class="styles__Illustration-sc-dzx1ao-4 hgYHjx"
    src="/img/hero-illustration.svg"
  />
</main>
`;
